### PR TITLE
startUpdate require wait stage

### LINKED ISSUE #1403: [Feature Request] Make startUpdate users aware that it's synchronous w/ worker
### Is your feature request related to a problem? Please describe.

See https://github.com/temporalio/features/issues/469

### Describe the solution you'd like

https://github.com/temporalio/sdk-typescript/blob/e41596bf3d78852c342f3f36b4ea21514f133c21/packages/client/src/workflow-client.ts#L162
Shouldn't be optional to pass what you'll wait for.

### PR DESCRIPTION
**Pull request by dandavison**
Closes #1403

## What was changed
Make `startUpdate` require
```typescript
waitForStage: WorkflowUpdateStage.ACCEPTED
```

## Why?
We decided to require this to make it explicit that the call -- despite being named `startXXX` -- blocks until a worker is available and accepts/rejects the update.

### CODE REVIEW

File: packages/client/src/workflow-client.ts [hunk 1]
```diff
@@ -138,30 +138,49 @@ export interface WorkflowHandle<T extends Workflow = Workflow> extends BaseWorkf
...
    *
    * @param def an Update definition as returned from {@link defineUpdate}
-   * @param options Update arguments
+   * @param options update arguments, and update lifecycle stage to wait for
+   *
+   * Currently, startUpdate always waits until a worker is accepting tasks for the workflow and the
+   * update is accepted or rejected, and the options object must be at least
+   * ```ts
+   * {
+   *   waitForStage: temporal.api.enums.v1.UpdateWorkflowExecutionLifecycleStage.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ACCEPTED
```
Comment (dandavison) [inline packages/client/src/workflow-client.ts:155]: TODO: offer a more ergonomic way to import this value

Comment (mjameswh) [reply packages/client/src/workflow-client.ts:155]: We need to fix this before we merge this PR. Fixing this later would be API breaking.

Look at `ScheduleOverlapPolicy` in schedule-client.ts/schedule-helpers.ts/schedule-types.ts, or `VersioningIntent` in `versioning-intent-enum.ts`. I'm ok with either approaches.

Comment (dandavison) [reply packages/client/src/workflow-client.ts:155]: Agreed, this is a blocking TODO (I should have marked the PR as draft).

**Comment by dandavison** [comment]
This is ready for review now.

https://github.com/temporalio/features/pull/504 updates the features repo to be compatible with the backwards-incompatible SDK changes made in this PR.

File: packages/client/src/workflow-update-stage.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { temporal } from '@temporalio/proto';
+import { checkExtends } from '@temporalio/common/lib/type-helpers';
+
+export enum WorkflowUpdateStage {
+  /** This is not an allowed value. */
+  UNSPECIFIED = 0,
```
Comment (mjameswh) [inline packages/client/src/workflow-update-stage.ts:6]: Any reason not to omit that value?

Comment (mjameswh) [reply packages/client/src/workflow-update-stage.ts:6]: Ah, forget it. It's required because of the `checkExtends` below. Fine.

**Review by mjameswh — APPROVED** [review]
Please fix the lint issue.

Also, don't merge just yet. I need to stabilize CI.

**Pull request merged**